src/datatypes/src/schema.rs

```diff
@@ -43,6 +44,8 @@ pub struct Schema {
     /// Timestamp key column is the column holds the timestamp and forms part of
     /// the primary key. None means there is no timestamp key column.
     timestamp_index: Option<usize>,
+    /// Columns default [`Expr`]
+    column_defaults: HashMap<String, Expr>,
```

Comment:
Just put this field into `Schema` to avoid building it every time. @evenyag .

Reply:
We want to keep the schema struct simple and we usually don't have to read `column_defaults`. Also, a schema may not depend on an `Expr`.

Datafusion only computes the default value once and stores it in the projection plan.
https://github.com/apache/datafusion/blob/fe66daaa81738dc60afc297a79897faaaa127724/datafusion/sql/src/statement.rs#L1457-L1480

Reply:
Yes, but we don't have a plan cache yet, so the computation will be repeated every time, it's not necessary at all.

Reply:
But looks like we need to create it every time because the `current_timestamp` must return the current time.

Reply:
I don't think schema is a proper place to cache this. It also adds the whole `datafusion-expr` to the dependencies of a basic data type crate.

We can cache the default value in the `Table`.
https://github.com/GreptimeTeam/greptimedb/blob/03c933c0069994270ba4084523df70867954142a/src/table/src/table.rs#L35-L40
We have a cache for the `Table`.
https://github.com/GreptimeTeam/greptimedb/blob/03c933c0069994270ba4084523df70867954142a/src/catalog/src/kvbackend/table_cache.rs#L29-L30

What do you think? @waynexia

Reply:
We should not cache at all, because the `create_default` always returns the current timestamp when it's called. It's not expected.

Reply:
The issue is that `get_column_default` returns a reference like `Option<&Expr>`. Consequently, I need to compute the `column_defaults` each time I create the adapters, even if `get_column_default` won't be used at all.

I don't know if there is a better way @waynexia . Maybe we can proposal the Datafusion to change this function signature.

Reply:
> We should not cache it at all, because the create_default always returns the current timestamp when it's called. If we cache it, the default value will not be changed after that. It's not expected.

Maybe we should document this while creating the adapter.

Reply:
>The issue is that get_column_default returns a reference like Option<&Expr>. Consequently, I need to compute the column_defaults each time I create the adapters, even if get_column_default won't be used at all.

For situations that the default value is a function like `current_timestamp`, maybe we should return that function in `Option<&Expr>` instead of literal value? Like `Expr::ScalarFunction(current_timestamp)` instead of `Expr::Literal(12345)`

Reply:
Fixed.